### PR TITLE
BUG: Fix DOP853 error norm for complex problems

### LINKED ISSUE #11260: BUG: DOP853 with complex data computes complex error norm, causing step failure
I find that the new `DOP853` method for a problem in the complex domain raises a `TypeError` when trying to compute `min_step` [here](https://github.com/scipy/scipy/blob/master/scipy/integrate/_ivp/rk.py#L120). It seems to me that `DOP853._estimate_error_norm` (in contrast to `RungeKutta._estimate_error_norm`) may return a complex number, leading to a complex increment in `t`.

I suspect the L_2 norms computed in [here](https://github.com/scipy/scipy/blob/master/scipy/integrate/_ivp/rk.py#L500) just need to be computed with the same `norm` function that `RungeKutta` calls (the problem being that `abs` isn't called before squaring).

#### Reproducing code example:

```
import numpy as np
from scipy.integrate._ivp.rk import DOP853, RK45

def f(t, y):
    return - 3 * y

dop = DOP853(t0=0, t_bound=1, fun=f, y0=np.array([1j]))

print(dop._estimate_error_norm(dop.K, 1, np.ones(1)))
dop.step()
dop.step()
print(dop.t)
dop.step()
```

prints two complex numbers (for the error norm and for `t` after two timesteps) on my laptop. Calling `step` the third time leads to the error below, which does not occur upon replacing `DOP853` above with, say, `RK45`.

#### Error message:

```
TypeError                                 Traceback (most recent call last)
<ipython-input-123-accff485a0c9> in <module>
     13 dop.step()
     14 print(dop.t)
---> 15 dop.step()

~/miniconda3/lib/python3.7/site-packages/scipy/integrate/_ivp/base.py in step(self)
    180         else:
    181             t = self.t
--> 182             success, message = self._step_impl()
    183 
    184             if not success:

~/miniconda3/lib/python3.7/site-packages/scipy/integrate/_ivp/rk.py in _step_impl(self)
    118         atol = self.atol
    119 
--> 120         min_step = 10 * np.abs(np.nextafter(t, self.direction * np.inf) - t)
    121 
    122         if self.h_abs > max_step:

TypeError: ufunc 'nextafter' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```

#### Scipy/Numpy/Python version information:

```
1.4.0 1.17.3 sys.version_info(major=3, minor=7, micro=3, releaselevel='final', serial=0)
```

### LINKED ISSUE #11972: BUG (solved): Error estimation in DOP853 ODE solver fails for complex domain
Solving an ODE in a complex domain using the `DOP853` solver fails as the function `_estimate_error_norm` generally returns complex results that eventually propagate into the variable `t` which must remain real. This leads to the following error:

```
File "/u/fs1/as2372/.local/lib/python3.6/site-packages/scipy-1.4.1-py3.6-linux-x86_64.egg/scipy/integrate/_ivp/ivp.py", line 577, in solve_ivp
    message = solver.step()
  File "/u/fs1/as2372/.local/lib/python3.6/site-packages/scipy-1.4.1-py3.6-linux-x86_64.egg/scipy/integrate/_ivp/base.py", line 182, in step
    success, message = self._step_impl()
  File "/u/fs1/as2372/.local/lib/python3.6/site-packages/scipy-1.4.1-py3.6-linux-x86_64.egg/scipy/integrate/_ivp/rk.py", line 120, in _step_impl
    min_step = 10 * np.abs(np.nextafter(t, self.direction * np.inf) - t)
TypeError: ufunc 'nextafter' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```

The issue is caused by lines 500-501 of `rk.py` that do NOT evaluate the squared norms of their respective vectors, and should be changed to

```
        err5_norm_2 = np.sum(np.abs(err5)**2)
        err3_norm_2 = np.sum(np.abs(err3)**2)
```

or equivalent.

### PR DESCRIPTION
**Pull request by peterbell10**
#### Reference issue
Fixes gh-11972, fixes gh-11260

#### What does this implement/fix?
Use `np.linal.norm(x)**2` instead of `np.sum(x**2)` for complex number support. I also noticed the tests in `_ivp` weren't being run when using `runtests.py` because the tests aren't packaged correctly.

### CODE REVIEW

**Comment by ev-br** [comment]
This seems real:

```
[gw0] linux -- Python 3.6.7 /home/travis/virtualenv/python3.6.7/bin/python
/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/scipy/integrate/_ivp/tests/test_rk.py:36: in test_error_estimation_complex
    assert err_norm.dtype.kind == 'f'
E   AttributeError: 'int' object has no attribute 'dtype'
        err_norm   = 0
        h          = 0.2
        solver     = <scipy.integrate._ivp.rk.DOP853 object at 0x7f407c8d8048>
        solver_class = <class 'scipy.integrate._ivp.rk.DOP853'>
```

**Comment by ilayn** [comment]
also norm takes the square root of the sum but the code replaced doesn't have the square root. No idea if this is relevant though.

**Comment by peterbell10** [comment]
> AttributeError: 'int' object has no attribute 'dtype'

Looks like the test is hitting the zero-error case which doesn't return a numpy scalar. Moved to `np.isrealobj`.

> also norm takes the square root of the sum but the code replaced doesn't have the square root.

Not relevant to that test, but does explain the increase in `result.nfev` in `test_integration_complex`. Thanks.

File: scipy/integrate/_ivp/tests/test_ivp.py [hunk 1]
```diff
@@ -230,26 +230,26 @@ def test_integration_complex():
         assert_equal(res.status, 0)
 
         if method == 'DOP853':
-            assert_(res.nfev < 35)
```
Comment (nmayorov) [inline scipy/integrate/_ivp/tests/test_ivp.py:233]: The fact that `test_integration_complex` worked with strangely evaluated "complex norms" is not ideal. Maybe we should add a more substantial test for integration in complex domain.

As for this PR --- looks good to me.

**Pull request merged**